.github/workflows/ci.yaml

```diff
@@ -57,3 +57,22 @@ jobs:
 
         - name: Lint
           run: make lint
+
+  Generate:
+    steps:
+        - uses: actions/checkout@v4
+
+        - uses: actions/setup-go@v5
+          with:
+            go-version: '1.22'
+            cache: false
+
+        - name: Generate check
+          # Run go generate and see if anything changed.
+          run: |
+            go generate ./...
+            if [ -n "$(git status --porcelain)" ]; then
```

Comment:
This will fail if any files not added to git are created (non-ignored caches, etc.), which I think is stricter than we need here. I've always used `git diff --exit-code` for this.

Reply:
`go generate` doesn't necessarily just generate code, it can generate anything. Using `git diff --exit-code` we could miss cases where a generated resource had never been committed. That's why I used the porcelain option.

Currently this is working, and given that this job just runs` go generate`, I think it's reasonable to manually update `.gitignore` for any problems we encounter.